src/main.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+const (
+	githubContextEnv = "GITHUB_CONTEXT"
+	jobContextEnv    = "JOB_CONTEXT"
+)
+
+var rootCmd = func() cli.Command {
+	return &cli.RootCommand{
+		Name: "send-google-chat-webhook",
+		Commands: map[string]cli.CommandFactory{
+			"chat": func() cli.Command {
+				return &cli.RootCommand{
+					Name:        "workflownotification",
+					Description: "notification for workflow",
+					Commands: map[string]cli.CommandFactory{
+						"workflownotification": func() cli.Command {
+							return &WorkflowNotificationCommand{}
+						},
+					},
+				}
+			},
+		},
+	}
+}
+
+type WorkflowNotificationCommand struct {
+	cli.BaseCommand
+	flagWebhookUrl string
+}
+
+func (c *WorkflowNotificationCommand) Desc() string {
+	return "Send a message to a Google Chat space"
+}
+
+func (c *WorkflowNotificationCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  The chat command sends messages to Google Chat spaces.
+`
+}
+
+func (c *WorkflowNotificationCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+
+	f := set.NewSection("Chat space options")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "webhook-url",
+		Example: "https://chat.googleapis.com/v1/spaces/<SPACE_ID>/messages?key=<KEY>&token=<TOKEN>",
+		Default: "",
+		Target:  &c.flagWebhookUrl,
+		Usage:   `Webhook URL from google chat`,
+	})
+
+	return set
+}
+
+func (c *WorkflowNotificationCommand) Run(ctx context.Context, args []string) error {
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	args = f.Args()
+	if len(args) != 0 {
+		return fmt.Errorf("expected 0 arguments, got %q", args)
+	}
+
+	ghJsonStr := os.Getenv(githubContextEnv)
+	if ghJsonStr == "" {
+		return fmt.Errorf("environment var %s not set", githubContextEnv)
+	}
+	jobJsonStr := os.Getenv(jobContextEnv)
+	if jobJsonStr == "" {
+		return fmt.Errorf("environment var %s not set", jobContextEnv)
+	}
```

Comment:
Similar to `c.NewFlagSet()`, it would be good to update to the latest CLI package and use `c.GetEnv` instead. This will make testing easier. This can be a follow-up.